### ISSUE TITLE
Text menus versus Tool Shelf

### ISSUE THREAD

**Issue by ReinerBforartists** [Bforartists/Bforartists#737]
We have text menus operators. And we have the new tool shelf. The text menus contains old operators. The tool shelf contains new operators. They do the same. But acts different.

What is disturbing here is that you have operators which are not in the tool shelf mixed with operators that are in the tool shelf. And the ones that are in the tool shelf is all double menu entries.

We need a way to hide the old operators away, but keep them reachable. Preferably having them close to the tool shelf tools. Like in a special legacy tab. Or we put it into so called legacy menus to keep them out of the way.

Tabs is a difficult chapter though with the new tool shelf. And is a own chapter. This task is a follow up once we have solved the tabs problem defined here: https://github.com/Bforartists/Bforartists/issues/325

Postponing ...

![toolshelftools](https://user-images.githubusercontent.com/13919702/55535467-a512fe00-56b7-11e9-86b7-124f01182242.jpg)

**Comment by Draise14** [comment]
I had this system sorta unified in a hotkey system, so you activate a legacy on drag of the one hotkey, and the interactive widget on normal key press. The visual feedback on transformations is awesome, and the same functionality between legacy and new tools being same/similar was unified in the same workflow. We have had users request for this in the past, transform = widget change. It makes sense the widget activates to the mode you move in. 

I personally would.. prefer all tools, legacy and new widget based ones, to be in the toolbar. It made sense in BFA1, would make sense in BFA2 - and the iconology would need to all be similar in the future. I would probably rename legacy operators to that, legacy, or as their specific tool that is differenent than the double entry. I would also then remove tools from header menus, as that too made sense in BFA1, and also minimized everything. Less dropdowns and navigation, more mouse over tool workflows - like the toolbar. 

And.. since you can add sub-menu tool and new tool buttons now, I would place them in their categories in the menus. Like legacy transform would go under the new transform icon as a subtool. I would try put things into subtools, because I know in trueSpace this worked, and it would not crowd the toolbar - and keeping them in sections also. No need to switch toolbars, but we'll see what they develop later in the source... 

But yeah, this would be a tool by tool case and.. lot's of work.  

*I got my electricity back though! Will get back to compiling and taking on some tasks next week.*

TL:DR
1. Move all legacy tools to toolshelf
2. Unify iconology
3. If double entry with essential functionality the same = put legacy as a sub tool of the new tool in shelf
4. If double entry is different in functinolaity but same name = change name of legacy tool, add as a new tool in shelf
5. Remove all operator tools from the header menu's, similar to BFA1
6. Create keymap that unifies same functionality tools that have immediate interaction and widget interaction to have the same hotkey just activated on drag (immediate).

**Comment by ReinerBforartists** [comment]
That's all a own task each, after we decided how to proceed here. First we need a tab system back in the tool shelf.

**Comment by ReinerBforartists** [comment]
This task is too general, and so not to fix. Catch them one by one. 

Closing.
